packages/shared/lib/clients/orchestrator.ts

```diff
@@ -669,6 +677,53 @@ export class Orchestrator {
         }
     }
 
+    async updateSyncFrequency({
+        syncId,
+        interval,
+        syncName,
+        environmentId,
+        activityLogId,
+        logCtx
+    }: {
+        syncId: string;
+        interval: string;
+        syncName: string;
+        environmentId: number;
+        activityLogId?: number;
+        logCtx?: LogContext;
+    }): Promise<Result<void>> {
+        const isGloballyEnabled = await featureFlags.isEnabled('orchestrator:schedule', 'global', false);
+        const isEnvEnabled = await featureFlags.isEnabled('orchestrator:schedule', `${environmentId}`, false);
+        const isOrchestrator = isGloballyEnabled || isEnvEnabled;
+
+        // Orchestrator
+        const scheduleName = this.getScheduleName({ environmentId, syncId });
+        const frequencyMs = ms(interval as StringValue);
+        const res = await this.client.updateSyncFrequency({ scheduleName, frequencyMs });
+
+        // Legacy
+        const { success, error } = await updateSyncScheduleFrequency(syncId, interval, syncName, environmentId, activityLogId, logCtx);
+
+        if (isOrchestrator) {
+            if (res.isErr()) {
+                errorManager.report(res.error, {
+                    source: ErrorSourceEnum.PLATFORM,
+                    operation: LogActionEnum.SYNC_CLIENT,
+                    environmentId,
+                    metadata: {
+                        syncName,
+                        scheduleName,
+                        frequencyMs
+                    }
+                });
+            } else {
+                await logCtx?.info(`Updated sync "${syncId}" frequency to ${frequencyMs}ms.`);
```

Comment:
syncId is already in the metadata, no entirely needed for customer facing message